submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionRequestFilesMongoPersistenceService.kt

```diff
@@ -0,0 +1,40 @@
+package ac.uk.ebi.biostd.persistence.doc.service
+
+import ac.uk.ebi.biostd.persistence.common.model.SubmissionRequestFile
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestFilesPersistenceService
+import ac.uk.ebi.biostd.persistence.doc.db.data.SubmissionRequestDocDataRepository
+import ac.uk.ebi.biostd.persistence.doc.db.repositories.SubmissionRequestFilesRepository
+import ac.uk.ebi.biostd.persistence.doc.model.DocSubmissionRequestFile
+import uk.ac.ebi.extended.serialization.service.ExtSerializationService
+import java.util.stream.Stream
+
+class SubmissionRequestFilesMongoPersistenceService(
+    private val extSerializationService: ExtSerializationService,
+    private val requestRepository: SubmissionRequestDocDataRepository,
+    private val requestFilesRepository: SubmissionRequestFilesRepository,
+) : SubmissionRequestFilesPersistenceService {
+    override fun upsertSubmissionRequestFile(file: SubmissionRequestFile) {
+        requestRepository.upsertSubmissionRequestFile(file)
+    }
+
+    override fun getSubmissionRequestFile(accNo: String, version: Int, filePath: String): SubmissionRequestFile {
+        return requestFilesRepository
+            .getByPathAndAccNoAndVersion(filePath, accNo, version)
+            .toSubmissionRequestFile()
+    }
+
+    override fun getSubmissionRequestFiles(
+        accNo: String,
+        version: Int,
+        startingAt: Int,
+    ): Stream<SubmissionRequestFile> {
+        return requestFilesRepository
+            .findAllByAccNoAndVersionAndIndexGreaterThan(accNo, version, startingAt)
+            .map { it.toSubmissionRequestFile() }
+    }
+
+    private fun DocSubmissionRequestFile.toSubmissionRequestFile(): SubmissionRequestFile {
```

Comment:
here I think we should rather use  `DocFile.toExtFile`

Reply:
sorry not following, `toExtFile` does not use fire. The advantages of having a Doc over a DBObject is that we can use spring to search and validate inners fields.